### PR TITLE
Make it possible to override the registry / the images used for pods when running make functests.

### PR DESCRIPTION
**Pull request by fedepaol**

### CODE REVIEW

File: hack/run-functests.sh [hunk 1]
```diff
@@ -32,7 +36,7 @@ mkdir -p "$TESTS_REPORTS_PATH"
 if [ "$CNF_TESTS_IMAGE" != "" ]; then
```
Comment (yuvalk) [inline hack/run-functests.sh:36]: CNF_TESTS_IMAGE can be derived from TEST_POD_IMAGES_REGISTRY/TEST_POD_CNF_TEST_IMAGE
same goes for DPDK_TESTS_IMAGE

Comment (fedepaol) [reply hack/run-functests.sh:36]: I don't get the `DPDK_TESTS_IMAGE` part

Comment (fedepaol) [reply hack/run-functests.sh:36]: Also, CNF_TESTS_IMAGE is used as driver to choose the containerized version, whereas overriding the other two is optional. I can remove that variable, but I would need to reintroduce a flag to drive the tests to run in containerized mode, not sure the change would add value.

Comment (fedepaol) [reply hack/run-functests.sh]: @yuvalk please take a look.
I had to introduce another variable anyway (TESTS_IN_CONTAINER) but we don't repeat the cnf_tests_image anymore and it's probably more clear than relying on the presence of the image variable to trigger the logic.

File: hack/run-functests.sh [hunk 2]
```diff
@@ -17,6 +17,14 @@ export TESTS_REPORTS_PATH="${TESTS_REPORTS_PATH:-/tmp/artifacts/}"
 export failed=false
 export failures=()
 
+#env variables needed for the containerized version
+export TEST_POD_IMAGES_REGISTRY="${TEST_POD_IMAGES_REGISTRY:-quay.io/openshift-kni/}"
+export TEST_POD_CNF_TEST_IMAGE="${TEST_POD_CNF_TEST_IMAGE:-cnf-tests:4.5}"
+export TEST_POD_DPDK_TEST_IMAGE="${TEST_POD_DPDK_TEST_IMAGE:-dpdk:4.5}"
+
+export CNF_TESTS_IMAGE=$TEST_POD_IMAGES_REGISTRY$TEST_POD_CNF_TEST_IMAGE
```
Comment (yuvalk) [inline hack/run-functests.sh:25]: this is disabling any (easy) way of running the tests locally
I guess we'd better introduce yet another env like RUN_LOCALLY
and replace the if in L#32

Comment (yuvalk) [reply hack/run-functests.sh:25]: oh sorry, I see you already did that with $TESTS_IN_CONTAINER, ok

Comment (fedepaol) [reply hack/run-functests.sh:25]: I already introduced a new variable which is `TESTS_IN_CONTAINER`
If you look at line 40 below, it does what you are saying but with a different default

**Pull request merged**